src/onnx/parse_hardmax.cpp

```diff
@@ -38,35 +38,57 @@ struct parse_hardmax : op_parser<parse_hardmax>
     instruction_ref parse(const op_desc& /*opd*/,
                           const onnx_parser& parser,
                           const onnx_parser::node_info& info,
-                          const std::vector<instruction_ref>& args) const
+                          std::vector<instruction_ref> args) const
     {
-        auto input_lens = args[0]->get_shape().lens();
-        auto input_type = args[0]->get_shape().type();
+        auto input      = args[0];
+        auto input_lens = input->get_shape().lens();
 
         // default axis value is -1 for opset 13
         int64_t axis = -1;
 
         // axis value is 1 for previous opset versions
         if(parser.opset_version < 13)
         {
-            MIGRAPHX_THROW("HARDMAX: Opset version below 13 not supported.");
+            axis = 1;
         }
 
         if(contains(info.attributes, "axis"))
         {
             axis = parser.parse_value(info.attributes.at("axis")).at<int>();
         }
 
-        auto indices = info.add_instruction(make_op("argmax", {{"axis", axis}}), args);
+        if(parser.opset_version < 13)
+        {
+            // input is coerced into a 2D matrix of size NxD
```

Comment:
This effectively means that the output will contain a single 1, at the location of the global max value--is that correct?

Reply:
It depends on the axis value, in the older version if the input shape is e.g. [3,2,2] and axis is 0, it will coerce it to [1,12] in which case we will indeed have only single 1. If axis is 1 (defult), it will coerce it to [3,4] in which case we will have 3 ones for each of the batches. As described https://onnx.ai/onnx/operators/onnx__Hardmax.html#hardmax-11, and implementation in python backend tests: https://github.com/onnx/onnx/blob/rel-1.7.0/onnx/backend/test/case/node/hardmax.py.
As for division by zero, I guess it could happen if one of the input lens is zero. Solution can be to calculate d in the same way as n is calculated:
size_t d = 1;
for(int i = axis; i < input_lens.size(); i++)
{
  d *= input_lens[i];
}

Reply:
@bpickrel  As for division by zero, I guess it could happen if one of the input lens is zero. After some analysis, I believe the best way would be to handle it as a special case scenario as in onnxruntime, as in that case number of elements() for the input shape would be zero. Therefore returning an output shape with same dimensions filled with zeroes would be the correct result for both versions.